### PR TITLE
90765

### PR DESCRIPTION
**Pull request by saimekala07**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [90765](https://devops.dchbx.org/redmine/issues/90765) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | Philip Burrowes |

### CODE REVIEW

File: components/benefit_sponsors/app/models/benefit_sponsors/benefit_packages/benefit_package.rb [hunk 1]
```diff
@@ -411,6 +411,7 @@ def cancel_member_benefits(delete_benefit_package: false, enroll_notify: false)
           enrollments = HbxEnrollment.by_benefit_package(self).where(family_id: family.id).show_enrollments_sans_canceled
           enrollments.each do |hbx_enrollment|
             if hbx_enrollment.may_cancel_coverage?
+              hbx_enrollment.update_attribute(:terminate_reason, "retroactive_canceled") if benefit_application.retroactive_canceled?
```
Comment (raghuramg) [inline components/benefit_sponsors/app/models/benefit_sponsors/benefit_packages/benefit_package.rb:414]: use
`  hbx_enrollment.terminate_reason = 'retroactive_canceled' 
`
this attribute saved in the next line hbx_enrollment.cancel_coverage!

Comment (raghuramg) [reply components/benefit_sponsors/app/models/benefit_sponsors/benefit_packages/benefit_package.rb:414]: can you also modify if conditions in the method like below.
```
hbx_enrollment.cancel_coverage!
hbx_enrollment.notify_enrollment_cancel_or_termination_event(enrollment_notify_flag(enroll_notify)) unless hbx_enrollment.inactive?
```

File: components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb [hunk 2]
```diff
@@ -772,21 +772,38 @@ module BenefitSponsors
       context "when enrollment is in auto renewing state" do
         let(:enr_state) {"auto_renewing"}
 
+        before do
```
Comment (raghuramg) [inline components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb:775]: replace this with 
`   let(:renewal_state) { :canceled }`

File: components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb [hunk 3]
```diff
@@ -772,21 +772,38 @@ module BenefitSponsors
...
+
+        it "should not persist retro cancel reason for canceled applications" do
+          expect(hbx_enrollment.reload.terminate_reason).to eq nil
+        end
       end
 
       context "when enrollment is in coverage selected state state" do
         let(:enr_state) {"coverage_selected"}
 
+        before :each do
```
Comment (raghuramg) [inline components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb:793]: replace this with

`let(:renewal_state) { :retroactive_canceled }`

**Review by raghuramg — CHANGES_REQUESTED** [review]
Code changes looks good.

Added small suggestions with regard to code refactor.

File: components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb (hunk 2 above)
Comment (saimekala07) [reply components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb]: updated

File: components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb (hunk 3 above)
Comment (saimekala07) [reply components/benefit_sponsors/spec/models/benefit_sponsors/benefit_packages/benefit_package_spec.rb]: updated

File: components/benefit_sponsors/app/models/benefit_sponsors/benefit_packages/benefit_package.rb (hunk 1 above)
Comment (saimekala07) [reply components/benefit_sponsors/app/models/benefit_sponsors/benefit_packages/benefit_package.rb]: updated "update_attribute" statement.

After enrollment canceled persisted, hbx_enrollment.inactive? returns false. will trigger for waived enrollmnets

**Review by raghuramg — APPROVED** [review]
Added retroactive_canceled state benefit application. Reviewed code and specs. Please promote this for testing.

**Pull request merged**